### PR TITLE
Basic implementation

### PR DESCRIPTION
**Pull request by richfitz**
This PR implements the interpolation support required for odin (previously in [this file, bundled in odin](https://github.com/mrc-ide/odin/blob/master/inst/js/interpolate.js)).

### CODE REVIEW

File: test/interpolate.test.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    it("returns change points correctly", () => {
+        for (let i = 0; i < x.length; ++i) {
+            expect(obj.eval(x[i])).toEqual(y[i]);
+        }
+    });
+
+    it("returns midpoints correctly", () => {
+        for (let i = 0; i < x.length - 1; ++i) {
+            expect(obj.eval(x[i] + 0.5)).toBeCloseTo(y[i]);
```
Comment (hillalex) [inline test/interpolate.test.ts]: not `toEqual`? (and same for below test)

File: README.md [hunk 2]
```diff
@@ -4,7 +4,13 @@
 [![build-and-test](https://github.com/reside-ic/interpolate/actions/workflows/ci.yml/badge.svg)](https://github.com/reside-ic/interpolate/actions/workflows/ci.yml)
 [![codecov.io](https://codecov.io/github/reside-ic/interpolate/coverage.svg?branch=main)](https://codecov.io/github/reside-ic/interpolate?branch=main)
 
-Function interpolation - piecewise constant, linear and spline interpolation.
+Function interpolation - piecewise constant, linear and spline interpolation. There are many interpolation packages out there. This one is tailored to our needs in [odin-js](https://github.com/mrc-ide/odin-js)
+
+* Support interpolating over several series at once (for a single domain)
+* Same interface for piecewise-constant, piecewise-linear and spline interpolation
+* Same behaviour as R's `approx` function so that output matches other [odin](https://github.com/mrc-ide/odin) targets
```
Comment (hillalex) [inline README.md:11]: is there a way to cross-validate the results given by different targets? is that done in `odin`?

File: test/interpolate.test.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
+import {interpolateSearch} from "../src/base";
+import {InterpolatorConstant} from "../src/constant";
+import {InterpolatorLinear} from "../src/linear";
+import {InterpolatorSpline, solve} from "../src/spline";
+
+// look in odin:tests/testthat/test-js-support-interpolate.R for test cases
+describe("interpolateSearch can find points", () => {
+    it("Can find point", () => {
```
Comment (hillalex) [inline test/interpolate.test.ts]: I think there's a better name for this test perhaps, since it's only testing whether you can find limit points?

**Review by hillalex — COMMENTED** [review]
Looks good to me, in so far as I understand it!

File: test/interpolate.test.ts (hunk 1 above)
Comment (richfitz) [reply test/interpolate.test.ts]: you're right, should be that

File: README.md (hunk 2 above)
Comment (richfitz) [reply README.md:11]: I've put reference cases in the tests and linked back to the code that generated them. There's a test of equivalence in odin yes

**Review by hillalex — APPROVED** [review]

**Pull request merged**